src/interface/report/CombatLogParserContext.tsx

```diff
@@ -0,0 +1,26 @@
+import CombatLogParser from 'parser/core/CombatLogParser';
+import { createContext, ReactNode, useContext } from 'react';
+
+export interface CombatLogParserContext {
+  combatLogParser: CombatLogParser;
+}
+
+// This starts off undefined as we don't have an instance of CombatLogParser to work with
+// until it gets provided by the Provider.
+export const CombatLogParserCtx = createContext<CombatLogParserContext | undefined>(undefined);
+
+interface Props {
+  children: ReactNode;
+  combatLogParser: CombatLogParser;
+}
+export const CombatLogParserProvider = ({ children, combatLogParser }: Props) => (
+  <CombatLogParserCtx.Provider value={{ combatLogParser }}>{children}</CombatLogParserCtx.Provider>
+);
+
+export const useCombatLogParser = () => {
+  const context = useContext(CombatLogParserCtx);
+  if (context === undefined) {
+    throw new Error('useCombatLogParser must be within CombatLogParserProvider');
+  }
+  return context;
+};
```

Comment:
we may want to move the `Guide`'s `useAnalyzer` hook over here. its been quite useful.

(not for this PR tho, just a note for later)

Reply:
I have another PR where I'm adding a lot of contexts, that would probably fit very well with that PR.